### PR TITLE
Error handler middleware

### PR DESCRIPTION
**Pull request by crisu83**
Add support for error handler middleware.

### CODE REVIEW

**Comment by Jalle19** [comment]
I pushed some minor cleanups. @crisu83 what should we do about errors that occur before the main execution phase?

* schema validation errors
* query parsing errors
* query validation errors
* createContext() errors

**Comment by crisu83** [comment]
@Jalle19 I guess we could handle those in the same manner with some minor refactoring. We could actually make this a non-breaking change by some additional refactoring. We just need a `CallableMiddleware` that is used to wrap any `callable` that is passed in as the error handler.

File: src/GraphQL.php [hunk 1]
```diff
@@ -249,18 +252,82 @@ public static function execute(
...
+     * @param null                       $contextValue
+     * @param array                      $variableValues
+     * @param null|string                $operationName
+     * @param callable|null              $fieldResolver
+     * @param ErrorHandlerInterface|null $errorHandler
+     * @return ExecutionResult
+     * @throws InvariantException
+     * @throws SyntaxErrorException
+     */
+    public static function process(
```
Comment (Jalle19) [inline src/GraphQL.php]: Why move this to a static method?

Comment (crisu83) [reply src/GraphQL.php]: This wasn't a part of the `GraphQL` class at all before, only the public API.

File: src/api.php [hunk 2]
```diff
@@ -162,44 +164,33 @@ function printNode(NodeInterface $node): string
...
-        return (new ExecutionResult([], [$error]))->toArray();
-    }
-
-    $validationErrors = validate($schema, $document);
-    if (!empty($validationErrors)) {
-        return (new ExecutionResult([], $validationErrors))->toArray();
-    }
-
-    $result = execute(
+    $result = GraphQL::process(
```
Comment (Jalle19) [inline src/api.php]: This now doesn't go through the container, it used to do that.

Comment (Jalle19) [reply src/api.php]: Actually never mind, I read this wrong

**Comment by Jalle19** [comment]
@crisu83 this should be good to go, it's up to you if you want to fix the null check

**Comment by Jalle19** [comment]
Oh yeah and please update the documentation

File: src/Execution/Execution.php [hunk 3]
```diff
@@ -58,9 +58,9 @@ public function execute(
             });
         }
 
-        if (null !== $errorHandler) {
+        if ($errorHandler instanceof ErrorHandlerInterface) {
```
[the remark this answers is not in the document]
Comment (Jalle19) [reply src/Execution/Execution.php]: Pushed some minor changes

**Pull request merged**